### PR TITLE
Don't tree shake in `delete_pattern()`

### PR DESCRIPTION
**Pull request by kienstra**
### Summary of changes
A much simpler [subset](https://github.com/studiopress/pattern-manager/pull/192/files#r1227497186) of my bloated  https://github.com/studiopress/pattern-manager/pull/192 😄 

Maybe we'll consider https://github.com/studiopress/pattern-manager/pull/192, but if this PR doesn't work, I'll have to rework that.

### How to test

Phil's [testing steps](https://github.com/studiopress/pattern-manager/pull/192#issue-1744354622)

I couldn't reproduce it, so this PR might not fix it.

### CODE REVIEW

File: wp-modules/pattern-data-handlers/pattern-data-handlers.php [hunk 1]
```diff
@@ -323,10 +323,7 @@ function update_pattern( $pattern ) {
 function delete_pattern( string $pattern_name ): bool {
 	$wp_filesystem = \PatternManager\GetWpFilesystem\get_wp_filesystem_api();
 	$pattern_path  = get_pattern_path( $pattern_name );
-	$result        = $wp_filesystem && $wp_filesystem->exists( $pattern_path ) && $wp_filesystem->delete( $pattern_path );
-	tree_shake_theme_images( $wp_filesystem, 'copy_dir' );
```
Comment (kienstra) [inline wp-modules/pattern-data-handlers/pattern-data-handlers.php:327]: I shouldn't have called `tree_shake_theme_images()` in `delete_pattern()`.

It's not needed here. There are 2 other places that call it.

Hiding it in `delete_pattern()` probably caused this bug, and invites further bugs.

File: wp-modules/api-data/api-data.php [hunk 2]
```diff
@@ -108,6 +110,7 @@ function get_pattern_names() {
  */
 function delete_pattern( $request ) {
 	$is_success = \PatternManager\PatternDataHandlers\delete_pattern( $request->get_params()['patternName'] );
+	tree_shake_theme_images( get_wp_filesystem_api(), 'copy_dir' );
```
Comment (kienstra) [inline wp-modules/api-data/api-data.php:113]: Runs on deleting a pattern in the Patterns UI:

Screenshot 2023-06-13 at 1 19 55 AM

This has a lower risk of being a whack-a-mole bug.

File: wp-modules/pattern-data-handlers/pattern-data-handlers.php (hunk 1 above)
Comment (kienstra) [reply wp-modules/pattern-data-handlers/pattern-data-handlers.php:327]: Like Phil [pointed out](https://github.com/studiopress/pattern-manager/pull/192/commits/d33a28b109f69ff3e079a4cd5aafa180672d5e2d#diff-8ba31aa77fc662d19b7f20a487d3c2e0d83ac9ab99b843479f7ddae96b0649edR147-R1530), this made the order of `delete_pattern()` calls important.

The `rest_after_insert` [callback](https://github.com/studiopress/pattern-manager/blob/d33a28b109f69ff3e079a4cd5aafa180672d5e2d/wp-modules/editor/model.php#L83) runs after [delete_pattern()](https://github.com/studiopress/pattern-manager/blob/d33a28b109f69ff3e079a4cd5aafa180672d5e2d/wp-modules/editor/model.php#L151). 

So we shouldn't need to tree-shake in `delete_pattern()`. The `rest_after_insert` callback will do that.

File: wp-modules/app/js/src/components/VersionControlNotice/index.tsx [hunk 3]
```diff
@@ -29,8 +29,6 @@ export default function VersionControlNotice( {
 					div: <div style={ { marginTop: '1rem' } }></div>,
 					a: (
 						<a
-							target="_blank"
-							rel="noreferrer"
```
Comment (kienstra) [inline wp-modules/app/js/src/components/VersionControlNotice/index.tsx:33]: Unrelated, just to get linting to pass

File: wp-modules/pattern-data-handlers/pattern-data-handlers.php (hunk 1 above)
Comment (kienstra) [reply wp-modules/pattern-data-handlers/pattern-data-handlers.php:327]: The `rest_after_insert` [callback](https://github.com/studiopress/pattern-manager/blob/d33a28b109f69ff3e079a4cd5aafa180672d5e2d/wp-modules/editor/model.php#L83) runs after [delete_pattern()](https://github.com/studiopress/pattern-manager/blob/d33a28b109f69ff3e079a4cd5aafa180672d5e2d/wp-modules/editor/model.php#L151). 

So we shouldn't need to tree-shake in `delete_pattern()`. The `rest_after_insert` callback will do that.

**Comment by johnstonphilip** [comment]
This fixes the issue for me. I am good to merge this as-is :)

**Comment by kienstra** [comment]
Thanks a lot for reviewing this, Phil!

**Pull request merged**